### PR TITLE
Dev eve output redis async v10

### PR DESCRIPTION
**Pull request by fooinha**
* commit ( documentation )
  * doc: async: true ## if redis replies are read asynchronously

* commit ( code )
  * eve: detects libevent for async redis at configure
  * eve: moves redis output code to new file - util-log-redis.{c,h}
  * eve: redis ECHO and QUIT commands for async mode
  * eve: redis output defaults if conf is missing

Replaces #2636 

Many thanx to @jasonish for endless support when testing this. Please review.

----

Removed:
  *  eve: redis pipeline batch commands with async mode

### CODE REVIEW

**Comment by inliniac** [comment]
@jasonish I'll wait for your ACK/NACK before moving forward. Thanks!

File: src/util-error.h [hunk 1]
```diff
@@ -328,6 +328,7 @@ typedef enum {
     SC_ERR_VAR_LIMIT,
     SC_WARN_DUPLICATE_OUTPUT,
     SC_WARN_CHMOD,
+    SC_ERR_NO_LIBEVENT
```
Comment (jasonish) [inline src/util-error.h]: Would this be better named SC_ERR_NO_REDIS_ASYNC maybe?

File: src/util-logopenfile.c [hunk 2]
```diff
@@ -210,7 +211,7 @@ static PcieFile *SCLogOpenPcieFp(LogFileCtx *log_ctx, const char *path,
                                  const char *append_setting)
 {
 #ifndef __tile__
-    SCLogError(SC_ERR_INVALID_YAML_CONF_ENTRY, 
+    SCLogError(SC_ERR_INVALID_YAML_CONF_ENTRY,
```
Comment (jasonish) [inline src/util-logopenfile.c]: This doesn't belong to this feature.

**Comment by jasonish** [comment]
Other than the 2 very minor issues just noted, I'm now happy with this, and I'm unable to break it.

Future consideration: If async is enabled, use it by default.

**Comment by jasonish** [comment]
I rebased to master and ran through Prscript:
- PR jasonish-pcap: https://buildbot.openinfosecfoundation.org/builders/jasonish-pcap/builds/115
- PR jasonish: https://buildbot.openinfosecfoundation.org/builders/jasonish/builds/467

File: src/util-error.h (hunk 1 above)
Comment (fooinha) [inline src/util-error.h]: Changed this ... same PR.

File: src/util-logopenfile.c (hunk 2 above)
Comment (fooinha) [inline src/util-logopenfile.c]: Space is back ... same PR

**Comment by inliniac** [comment]
Please don't update existing PRs/branches. Jason ran PRscript but now it's unclear if it ran on the latest code.

**Comment by fooinha** [comment]
Sorry, creating a new branch and PR ...

**Pull request closed without merging**